### PR TITLE
Changes to make identify work

### PR DESCRIPTION
**Pull request by puneet-pdx**
Related to issue: # https://devtopia.esri.com/runtime/kotlin/issues/4584

### Description:

Added changes to make Identify work

### Summary of changes:

- Added graphicsOverlay and mapVeiwProxy to TraceState
- Added `selectedGeoElementsAsStartingPoints: StateFlow<List<GeoElement>>` to TraceState to keep track of SelectedGeoElements
- Added method to perform identify and to add tap location graphic on the screen
- Added Trace button to a Column outside to LazyColumn to make it non scrollable

### Pre-merge Checklist

- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/) Job for this PR has been run
  - [x] link: https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/96/
- Unit and/or integration tests have been added to exercise this PR's logic, and the tests are passing:
  - [ ] Yes
  - [ ] No

### CODE REVIEW

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt [hunk 1]
```diff
@@ -66,6 +78,16 @@ private class TraceStateImpl(
...
 
+    private val _selectedGeoElementsAsStartingPoints: MutableStateFlow<MutableList<GeoElement>> =
+        MutableStateFlow(
+            mutableListOf()
+        )
+    override val selectedGeoElementsAsStartingPoints: StateFlow<List<GeoElement>> =
+        _selectedGeoElementsAsStartingPoints.asStateFlow()
+
+    private lateinit var tapPoint: Point
+    private var screenPoint: ScreenCoordinate? = null
```
Comment (sorenoid) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt:89]: It doesn't seem like either of these are needed as properties if both are passed from `addStartingPoint` to the `identify` method? 

If there is a reason to keep them, consider not having one as lateinit and the other as nullable. I would consider a computed getter on `screenPoint` to derive its value from the current value of `tapPoint` as follows

```kotlin
private var tapPoint: Point? = null
private val screenPoint: ScreenCoordinate
    get() = tapPoint?.let { mapViewProxy.locationToScreen(it) } ?: throw IllegalStateException("tapPoint cannot be null")
```

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceOptionsScreen.kt [hunk 2]
```diff
@@ -100,25 +100,27 @@ internal fun TraceOptionsScreen(
...
+                    )
+                }
+                item {
+                    StartingPointsEditor(onAddStartingPointButtonClicked)
                 }
+                item {
+                    AdvancedOptions()
+                }
+            }
+            Button(onClick = { onPerformTraceButtonClicked() }) {
```
Comment (sorenoid) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceOptionsScreen.kt:122]: this is a nice improvement 👍 .

**Review by sorenoid — CHANGES_REQUESTED** [review]
nice improvement. One suggestion to consider.

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt (hunk 1 above)
Comment (puneet-pdx) [reply toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt:89]: yep, not needed as properties. Updated

**Review by sorenoid — APPROVED** [review]
👍

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt [hunk 3]
```diff
@@ -125,17 +149,43 @@ public class TraceState(
      * A single tap handler to identify starting points on the map. Call this method
      * from [com.arcgismaps.toolkit.geoviewcompose.MapView] onSingleTapConfirmed lambda.
      *
```
Comment (eri9000) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt:151]: doc needs to be updated

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceOptionsScreen.kt [hunk 4]
```diff
@@ -100,25 +100,27 @@ internal fun TraceOptionsScreen(
...
-            item {
-                AdvancedOptions()
-            }
-            item {
-                Button(onClick = { onPerformTraceButtonClicked() }) {
-                    Text(stringResource(id = R.string.trace))
+        Column (horizontalAlignment = Alignment.CenterHorizontally) {
+            LazyColumn(
+                modifier = Modifier
+                    .padding(3.dp)
```
Comment (eri9000) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceOptionsScreen.kt:106]:
```suggestion
                    .padding(10.dp)
```
```suggestion
                    .padding(3.dp)
```

**Review by eri9000 — APPROVED** [review]
👍🏼

**Pull request merged**